src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -92,10 +108,8 @@ private void Enqueue(float value)
                 {
                     // It is possible that two threads both think the buffer is full, but only one get to actually flush it, the other
                     // will eventually enter this code path and potentially calling Flushing on a buffer that is not full, and that's okay too.
-                    lock (_bufferedValues)
-                    {
+                    lock (this) // Lock the counter
```

Comment:
> I feel that desire not to lock on 'this' stemmed from code access security days, when we were trying to prevent issues from active hostile clients

FWIW, that was never the reason I avoided it.  By locking on `this`, I may end up unexpectedly causing unnecessary contention with consuming code that happens to be locking on my object for its own synchronization.  And since I've inherently exposed the lock object, I may also be more subject to deadlocks, as I can no longer completely control lock ordering in cases where I have multiple locks I've needed to establish into a lock hierarchy.

It may not matter in this particular case.  I just have an inherent bias against the practice, completely unrelated to CAS.